hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/ECUnderReplicationHandler.java

```diff
@@ -153,37 +166,48 @@ public Map<DatanodeDetails, SCMCommand<?>> processAndCreateCommands(
         }
       }
       List<Integer> missingIndexes = replicaCount.unavailableIndexes(true);
+      Map<Integer, Pair<ContainerReplica, NodeStatus>> sources =
+              filterSources(replicas, deletionInFlight);
+      List<DatanodeDetails> nodes =
+              sources.values().stream().map(Pair::getLeft)
```

Comment:
I'm confused about this piece of code, but I may be missing something obvious.

There are two filters:

```
.filter(datanodeDetails ->
                              datanodeDetails.getPersistedOpState() ==
                              HddsProtos.NodeOperationalState.IN_SERVICE)
```

And

```
.filter(DatanodeDetails::isDecomissioned)
```
Are these not mutually exclusive? Ie, a node cannot be IN_SERVICE and isDecommissioned - so will this not filter everything out?

Inside filter sources, we already sort the IN_SERVICE above other states - I am wondering why we need to filter again here?